### DISCUSSION TITLE
Is it possible to pass the Connector hook from wagmi over the windows postMessage API?

### DISCUSSION THREAD

**Discussion by izuchukwu-eric** [wevm/wagmi#2555] [Q&A]
I currently trying to pass the Connector hook from wagmi over the window postMessage API, but i keep getting the below error: I would like to know if this actually possible? and is the Connector hook serializable? I have also tried converting the circular structure to JSON but no luck yet

![testhost-3000](https://github.com/wagmi-dev/wagmi/assets/57435925/6e023692-e084-4e79-9bed-847227f87bb7)

Here is what my code looks like

The types file:

```bash
import { Connector } from 'wagmi/connectors';

export interface ConnectedWallet {
    walletName: string;
    account: string | `0x${string}`;
    chainId: number;
    connector: Connector;
}

export interface MessagedWallet extends ConnectedWallet {
    requestSignature?: boolean;
}
```
Where i call postMessage:

```bash
import { MessageWallet } from "./types"

function postMessage(data: API) {
  if (typeof document === 'undefined') return

  const iframeElement = document?.getElementById(iframeId) as HTMLIFrameElement

  iframeElement?.contentWindow?.postMessage(data, '*')
}

function trySignIn(wallet?: MessagedWallet) {
  postMessage({ target: 'sign_in', data: wallet || null })
}
```

would appreciate any tip on how  to go about this

**Comment by okwme** [comment]
any updates here?
